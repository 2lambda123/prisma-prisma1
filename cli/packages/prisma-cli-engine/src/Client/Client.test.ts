import { Client } from './Client'
import { Config } from '../Config'
import { Environment } from 'prisma-yml'
import { Output } from '../index'

import * as opn from 'opn'
import { GraphQLClient } from 'graphql-request'
jest.mock('graphql-request')
jest.mock('opn')

test('throws when no cluster provided', async () => {
  const config = new Config()
  const env = new Environment(config.home)
  const output = new Output(config)
  const client = new Client(config, env, output)
  let error
  try {
    await client.listProjects()
  } catch (e) {
    error = e
  }

  expect(error).toMatchSnapshot()
})
it('should log that url could not be open and keep listening for login', async () => {
  GraphQLClient.mockImplementation(() => {
    return {
      request: () => {
        return {
          requestCloudToken: {
            secret: 'my-secret'
          }
        }
      },
    }
  })
  opn.mockImplementation(() => {
    return new Promise((resolve, reject) => {
      reject(new Error('Exited with code XY'));
			return;
		});
  })
  const config = new Config()
  const env = new Environment(config.home)
  const output = new Output(config)
  const client = new Client(config, env, output)
  try {
    const res = await client.login()
  } catch(err) {
    // irrelevant for this test
  }
  expect(output.stdout.output).toMatchSnapshot()
})