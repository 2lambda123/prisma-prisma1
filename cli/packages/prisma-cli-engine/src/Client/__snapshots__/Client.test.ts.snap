// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should log that url could not be open and keep listening for login 1`] = `
"Opening https://app.prisma.io/cli-auth?secret=my-secret in the browser

Could not open url. Please open https://app.prisma.io/cli-auth?secret=my-secret manually
"
`;

exports[`throws when no cluster provided 1`] = `[Error: No cluster set. Please set the "cluster" property in your prisma.yml]`;
